### PR TITLE
Allow installation with CuPy 10

### PR DESCRIPTION
**Pull request by grlee77**
Now that CuPy 10.0 is out, we should also allow cuCIM to be installed with CuPy 10.x in the next release.

I checked the integration repo and [it has the same constraint proposed here](https://github.com/rapidsai/integration/blob/a76cfc32e684f53cf0a9b4cb113cf0fd9fad51e7/conda/recipes/versions.yaml#L47-L48)

### CODE REVIEW

**Comment by grlee77** [comment]
okay, I guess `/conda/recipes/cucim/meta.yaml` had already been updated, so this is now just an update of `conda/environments/env.yml`

File: conda/environments/env.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: cucim
 channels:
   - conda-forge
 dependencies:
-  - cupy=9
+  - cupy>=9,<11.0.0a0
```
Comment (jakirkham) [inline conda/environments/env.yml:5]: Perhaps we should just drop this constraint? I think the original motivation was to ensure we got CuPy 9 as we needed the very latest features. Do things work without a constraint here?

Comment (grlee77) [reply conda/environments/env.yml:5]: I am fine with just keeping the lower bound. I think it should tend to find 10.0 without the constraint, but it doesn't hurt to indicate that it is incompatible with 8.x.

Comment (grlee77) [inline conda/environments/env.yml:5]:
```suggestion
  - cupy>=9
```

**Review by jakirkham — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]

**Comment by jakirkham** [comment]
@gpucibot merge

**Pull request merged**